backend/avro/communication/metadata/src/main/java/co/airy/avro/communication/MetadataRepository.java

```diff
@@ -0,0 +1,29 @@
+package co.airy.avro.communication;
+
+import co.airy.uuid.UUIDv5;
+
+import java.util.UUID;
+
+public class MetadataRepository {
+    public static Subject getSubject(Metadata metadataAction) {
+        final String subjectString = metadataAction.getSubject();
+        int lastIndexOf = subjectString.lastIndexOf(":");
+
+        // You do not have to pass an identifier if the namespace you want to
+        // use metadata for consists of a single object
+        if (lastIndexOf == -1) {
+            return new Subject(subjectString, null);
+        }
+
+        String namespace = subjectString.substring(0, lastIndexOf);
+        String identifier = subjectString.substring(lastIndexOf + 1);
+        return new Subject(namespace, identifier);
+    }
+
+    public static UUID getId(Metadata metadataAction) {
+        return UUIDv5.fromNamespaceAndName(metadataAction.getSubject(), metadataAction.getKey());
+    }
+    public static UUID getId(Subject subject, String key) {
+        return UUIDv5.fromNamespaceAndName(subject.toString(), key);
+    }
+}
```

Comment:
As we're introducing this class (big 👍 for it), I'm thinking it may be nicer long term if MetadataAction creation details are all here, meaning we don't construct the object manually but have "shortcuts" like `MetadataRepository.newConversationMetadata(key, value)` (naming can be improved, I'm sure) `MetadataRepository.newChannelMetadata` and so on.

That way we could "hide" the `type:id` convention here and make the caller code a little more compact/readable

Reply:
Will add these helpers.